modules/account-lib/src/coin/dot/utils.ts

```diff
@@ -31,4 +52,105 @@ export class Utils implements BaseUtils {
+  decodeCallMethod(
+    tx: string | UnsignedTransaction,
+    options: { metadataRpc: string; registry: TypeRegistry },
+  ): TransferArgs {
+    const { metadataRpc, registry } = options;
+    registry.setMetadata(createMetadata(registry, metadataRpc));
+    let methodCall: any;
+    if (typeof tx === 'string') {
+      try {
+        const payload = createTypeUnsafe(registry, 'ExtrinsicPayload', [tx, { version: EXTRINSIC_VERSION }]);
+        methodCall = createTypeUnsafe(registry, 'Call', [payload.method]);
+      } catch (e) {
+        methodCall = registry.createType('Extrinsic', polkaUtils.hexToU8a(tx), {
+          isSigned: true,
+        }).method;
+      }
+    } else {
+      methodCall = registry.createType('Call', tx.method);
+    }
+    const decodedArgs = methodCall.args[2].toJSON() as unknown as ProxyCallArgs;
+    return decodedArgs.args;
+  }
+
+  /**
+   * keyPairFromSeed generates an object with secretKey and publicKey using the polkadot sdk
+   * @param seed 32 bytes long seed
+   * @returns KeyPair
+   */
+  keyPairFromSeed(seed: Uint8Array): KeyPair {
+    const keyring = new Keyring({ type: 'ed25519' });
+    const keyringPair = keyring.addFromSeed(seed);
+    const pairJson = keyringPair.toJson();
+    const decodedKeyPair = decodePair('', base64Decode(pairJson.encoded), pairJson.encoding.type);
+    return new KeyPair({ prv: Buffer.from(decodedKeyPair.secretKey).toString('hex') });
+  }
+
+  /**
+   * Signing function. Implement this on the OFFLINE signing device.
+   *
+   * @param {KeyringPair} pair - The signing pair.
+   * @param {string} signingPayload - Payload to sign.
+   */
+  createSignedTx(pair: KeyringPair, signingPayload: string, transaction: UnsignedTransaction, options): string {
+    const { registry, metadataRpc } = options;
+    // Important! The registry needs to be updated with latest metadata, so make
+    // sure to run `registry.setMetadata(metadata)` before signing.
+    registry.setMetadata(createMetadata(registry, metadataRpc));
+    const { signature } = registry
+      .createType('ExtrinsicPayload', signingPayload, {
+        version: EXTRINSIC_VERSION,
+      })
+      .sign(pair);
+
+    // Serialize a signed transaction.
+    const txHex = construct.signedTx(transaction, signature, {
+      metadataRpc,
+      registry,
+    });
+    return txHex;
+  }
+
+  /**
+   * Returns the decoded keypair from a Dot keyring pair
+   *
+   * @param {KeyringPair} keyringPair
+   * @returns {DefaultKeys} default key format
+   */
+  decodeDotKeyringPair(keyringPair: KeyringPair): DefaultKeys {
+    if (keyringPair.isLocked) {
+      return {
+        pub: Buffer.from(keyringPair.publicKey).toString('hex'),
+      };
+    }
+    const keyPair = decodePair('', base64Decode(keyringPair.toJson().encoded), keyringPair.toJson().encoding.type);
+    return {
+      prv: Buffer.from(keyPair.secretKey).toString('hex').slice(0, 64),
+      pub: Buffer.from(keyPair.publicKey).toString('hex'),
+    };
+  }
 }
+
```

Comment:
Add verifySignature method:
Example: https://github.com/BitGo/BitGoJS/blob/master/modules/account-lib/src/coin/algo/utils.ts#L189

Reply:
https://polkadot.js.org/docs/util-crypto/examples/verify-signature

Verify signature these 3 args, we can't override base class function args in utils

Reply:
We need to find a way how to do that. Because this is a common and generic method used on wallet-platform.

Reply:
I can add in a Dot specific method to verifySignature in this class

Reply:
https://github.com/polkadot-js/common/blob/7047bf5aa513b029d878fb4f12625dffa4f37b30/packages/util-crypto/src/signature/verify.ts#L86

const { signatureVerify } = require('@polkadot/util-crypto');

verifySignature(message: Uint8Array, signature: Buffer, pub: Uint8Array): boolean {
	return signatureVerify(signedMessage, new UIntArray8(signature), hexPublicKey).isValid;
}

Reply:
Something like this I think you can do. Please let me know, also add tests for this util.